File: KIJUNG/WEEK_05/10866.cpp
#include <iostream>
#include <deque>
#include <string>

using namespace std;

string str;
int N;
int nums[20001];
int b = 10000;
int e = 10000;

int main(void){
    cin >> N;

    for(int i=0;i<N;i++){
        cin >> str;

        if(str == "push_front") cin >> nums[--b];
        else if(str == "push_back") cin >> nums[e++];
        else if(str == "pop_front") cout << ((e-b)? nums[b++]:-1) <<"aa\n";
        else if(str == "pop_back") cout << ((e-b)? nums[--e]:-1) <<"aa\n";
        else if(str == "size") cout << e-b << "aa\n";
        else if(str == "empty") cout << ((e-b)? 0:1) << "aa\n";
        else if(str == "front") cout << ((e-b)? nums[b]:-1) << "aa\n";
        else if(str == "back") cout << ((e-b)? nums[e-1]:-1) << "aa\n";
    }

    return 0;
}

File: DONGUK/WEEK 5/10866.py
from collections import deque
import sys

input = sys.stdin.readline

q = deque()

n = int(input())

for _ in range(n):
    s = input().rstrip().split()

    if s[0] == 'push_back':
        q.append(s[1])
    
    elif s[0] == 'push_front':
        q.appendleft(s[1])

    elif s[0] == 'pop_front':
        if len(q) == 0:
            print(-1)
        
        else:
            print(q.popleft())
    
    elif s[0] == 'pop_back':
        if len(q) == 0:
            print(-1)
        
        else:
            print(q.pop())

        
    elif s[0] == 'size':
        print(len(q))

    
    elif s[0] == 'empty':
        if len(q) == 0:
            print(1)
        
        else:
            print(0)

    elif s[0] == 'front':
        if len(q) == 0:
            print(-1)
        
        else:
            print(q[0])

    elif s[0] == 'back':
        if len(q) == 0:
            print(-1)
        
        else:
            print(q[-1])

    else:
        continue
